scripts/exglobal_diag.sh

```diff
@@ -176,100 +165,100 @@ EOFdiag
+                  if [[ "${CFP_MP:-"NO"}" = "YES" ]]; then
                      nm=$((nm+1))
                   fi
                else
-                  if [ $binary_diag = ".true." ]; then
-                     cat ${prefix}${type}_${loop}* > diag_${type}_${string}.${CDATE}${DIAG_SUFFIX}
+                  if [[ "${binary_diag}" = ".true." ]]; then
+                     cat "${prefix}${type}_${loop}"* > "diag_${type}_${string}.${CDATE}${DIAG_SUFFIX}"
                   else
-                     $CATEXEC -o diag_${type}_${string}.${CDATE}${DIAG_SUFFIX} ${prefix}${type}_${loop}*
+                     ${CATEXEC} -o "diag_${type}_${string}.${CDATE}${DIAG_SUFFIX}" "${prefix}${type}_${loop}"*
                   fi
                fi
                echo "diag_${type}_${string}.${CDATE}*" >> ${diaglist[n]}
                numfile[n]=$(expr ${numfile[n]} + 1)
-            elif [ $count -eq 1 ]; then
-                cat ${prefix}${type}_${loop}* > diag_${type}_${string}.${CDATE}${DIAG_SUFFIX}
-                if [ $DIAG_COMPRESS = "YES" ]; then
-                   $COMPRESS diag_${type}_${string}.${CDATE}${DIAG_SUFFIX}
+            elif [[ "${count}" -eq 1 ]]; then
+                cat "${prefix}${type}_${loop}"* > "diag_${type}_${string}.${CDATE}${DIAG_SUFFIX}"
+                if [[ "${DIAG_COMPRESS}" = "YES" ]]; then
+                   ${COMPRESS} "diag_${type}_${string}.${CDATE}${DIAG_SUFFIX}"
                 fi
                 echo "diag_${type}_${string}.${CDATE}*" >> ${diaglist[n]}
-                numfile[n]=$(expr ${numfile[n]} + 1)
+                numfile[n]=$(expr "${numfile[n]}" + 1)
             fi
          done
       done
-      echo $(date) END loop $string >&2
+      echo $(date) END loop "${string}" >&2
    done
 
    # We should already be in $DATA, but extra cd to be sure.
-   cd $DATA
+   cd "${DATA}" || exit
 
    # If requested, compress diagnostic files
-   if [ $DIAG_COMPRESS = "YES" -a $USE_CFP = "NO" ]; then
-      echo $(date) START $COMPRESS diagnostic files >&2
-      for file in $(ls diag_*${CDATE}${DIAG_SUFFIX}); do
-         $COMPRESS $file
+   if [[ "${DIAG_COMPRESS}" = "YES" && "${USE_CFP}" = "NO" ]]; then
+      echo $(date) START "${COMPRESS}" diagnostic files >&2
+      for file in $(ls diag_*"${CDATE}${DIAG_SUFFIX}"); do
+         ${COMPRESS} "${file}"
       done
-      echo $(date) END $COMPRESS diagnostic files >&2
+      echo $(date) END "${COMPRESS}" diagnostic files >&2
    fi
 
-   if [ $USE_CFP = "YES" ] ; then
-      chmod 755 $DATA/mp_diag.sh
-      ncmd=$(cat $DATA/mp_diag.sh | wc -l)
-      if [ $ncmd -gt 0 ]; then
+   if [[ "${USE_CFP}" = "YES" ]] ; then
+      chmod 755 "${DATA}/mp_diag.sh"
+      ncmd=$(cat "${DATA}/mp_diag.sh" | wc -l)
+      if [[ "${ncmd}" -gt 0 ]]; then
          ncmd_max=$((ncmd < max_tasks_per_node ? ncmd : max_tasks_per_node))
```

Comment:
Prefer if-else block
```suggestion
         if [[ ${ncmd} -lt ${max_tasks_per_node} ]]; then
            ncmd_max=${ncmd}
         else
            ncmd_max=${max_tasks_per_node}
         fi
```